### PR TITLE
Fix packages on linux don't show icon in desktop entry #136

### PR DESCRIPTION
**Pull request by develar**
Docs (we use OS X `icns` to generate linux icon set) will be added as part of `update readme` pull request.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,14 +1,13 @@
 node_modules/
-tmp/
```
Comment (develar) [inline .gitignore]: Otherwise `tmp.d.ts` is ignored. If it is still required, `/` maybe prepended to avoid such issues.

**Comment by stefanjudis** [comment]
There is a lot of stuff going on in your PR's. We have to definitely lower the entry barrier at some point. I don't think a lot of people can simply jump in and contribute.

Honestly I'm not really aware of what all these changes in 20 files do and it feels like a huge overhead...
Not sure how I can help here and how "me clicking the button" has any benefit. So go ahead if you wish. :)

**Comment by develar** [comment]
> can simply jump in and contribute.

Sorry. I don't see what is complex in my changes. 

`typings/main/ambient/gm/gm.d.ts` is type definitions (not written by me).

Maybe you mean precomputing of icons — promise in the package helper constructor and `await` in the `packageInDistributableFormat`. Yes, it adds complexity, but allows us:
* do not do it twice (number of linux architectures).
* do it in parallel (well, nodejs cannot to really perform parallel tasks, but at least IO).

Maybe you mean processing of OS X icons, but throwing error "Hi — you must double your work and repeat all icons in the yet another format for yet another OS" is not user friendly (also, Linux icons layout is complex (not all icons in one folder), so, in any case we have to somehow copy it in a smart way).

Also, OS X `icns`is not always contain 48x48 icon, so, we resize 128x128 icon.

Linux Icons "specification": http://stackoverflow.com/a/10981921

**Comment by stefanjudis** [comment]
> Sorry. I don't see what is complex in my changes.

I rather easy thing with 20 changed files. The diff is huge. I don't know what is generated. I don't know what all these files are for. The project lost me already. Sorry.

That's all fine, the only thing I'm saying is that we have to make it as easy as possible to people to jump in. Meaning powerful docs and an easy workflow. TypeScript makes it already harder so there is a lot of guidance needed. ;)

**Pull request merged**